src/libcore/iter/range.rs

```diff
@@ -325,37 +408,53 @@ impl<A: Step> Iterator for ops::RangeInclusive<A> {
 
     #[inline]
     fn size_hint(&self) -> (usize, Option<usize>) {
+        // This seems redundant with a similar comparison that `Step::steps_between`
+        // implementations need to do, but it separates the `start > end` case from `start == end`.
+        // `steps_between` returns `Some(0)` in both of these cases, but we only want to add 1
+        // in the latter.
```

Comment:
Should `steps_between` perhaps instead return something like `Result<Option<usize>, Unreachable>`?  Might make the documentation for the method easier to write, too.